src/main/java/io/github/genomicdatainfrastructure/discovery/filters/infrastructure/ckan/CkanFilterBuilder.java

```diff
@@ -120,9 +123,13 @@ private Filter filter(Map.Entry<String, CkanFacet> entry) {
                         .build())
                 .toList();
 
+        var type = metadata != null && metadata.type != null
+                ? metadata.type
+                : DEFAULT_FILTER_TYPE;
```

Comment:
To reduce the `null`s this could be written as:

```suggestion
        var type = Optional.ofNullable(metadata)
                .map(m -> m.type)
                .orElse(DEFAULT_FILTER_TYPE);
```